teammapper-backend/src/map/controllers/maps.gateway.ts

```diff
@@ -57,10 +59,12 @@ export class MapsGateway implements OnGatewayDisconnect {
   async onJoin(
     @ConnectedSocket() client: Socket,
     @MessageBody() request: IMmpClientJoinRequest
-  ): Promise<IMmpClientMap> {
+  ): Promise<IMmpClientMap | undefined> {
     const map = await this.mapsService.findMap(request.mapId)
-    if (!map)
-      return Promise.reject()
+    if (!map) {
+      this.logger.error(`Could not find map when client ${client.id} tried to join`);
```

Comment:
can you add the map id and the function name in the beginning? the class name is added by the  logger afaik but not the function name.